src/lib/optimizer/abstract_syntax_tree/ast_to_operator_translator.cpp

```diff
@@ -34,59 +35,14 @@
 #include "optimizer/abstract_syntax_tree/sort_node.hpp"
 #include "optimizer/abstract_syntax_tree/stored_table_node.hpp"
 #include "optimizer/abstract_syntax_tree/update_node.hpp"
+#include "optimizer/abstract_syntax_tree/validate_node.hpp"
 #include "utils/performance_warning.hpp"
 
 namespace opossum {
 
-// singleton
-ASTToOperatorTranslator& ASTToOperatorTranslator::get() {
-  static ASTToOperatorTranslator instance;
-  return instance;
-}
-
-ASTToOperatorTranslator::ASTToOperatorTranslator() {
-  /**
-   * Build a mapping from an ASTNodeType to a function that takes an ASTNode of such type and translates it into a set
-   * of operators and returns the root of them. We prefer this over a virtual AbstractASTNode::translate() call in order
-   * to keep the translation code in one place, i.e., this file.
-   */
-
-  // SQL operators
-  _operator_factory[ASTNodeType::StoredTable] =
-      std::bind(&ASTToOperatorTranslator::_translate_stored_table_node, this, std::placeholders::_1);
-  _operator_factory[ASTNodeType::Predicate] =
-      std::bind(&ASTToOperatorTranslator::_translate_predicate_node, this, std::placeholders::_1);
-  _operator_factory[ASTNodeType::Projection] =
-      std::bind(&ASTToOperatorTranslator::_translate_projection_node, this, std::placeholders::_1);
-  _operator_factory[ASTNodeType::Sort] =
-      std::bind(&ASTToOperatorTranslator::_translate_sort_node, this, std::placeholders::_1);
-  _operator_factory[ASTNodeType::Join] =
-      std::bind(&ASTToOperatorTranslator::_translate_join_node, this, std::placeholders::_1);
-  _operator_factory[ASTNodeType::Aggregate] =
-      std::bind(&ASTToOperatorTranslator::_translate_aggregate_node, this, std::placeholders::_1);
-  _operator_factory[ASTNodeType::Limit] =
-      std::bind(&ASTToOperatorTranslator::_translate_limit_node, this, std::placeholders::_1);
-  _operator_factory[ASTNodeType::Insert] =
-      std::bind(&ASTToOperatorTranslator::_translate_insert_node, this, std::placeholders::_1);
-  _operator_factory[ASTNodeType::Delete] =
-      std::bind(&ASTToOperatorTranslator::_translate_delete_node, this, std::placeholders::_1);
-  _operator_factory[ASTNodeType::DummyTable] =
-      std::bind(&ASTToOperatorTranslator::_translate_dummy_table_node, this, std::placeholders::_1);
-  _operator_factory[ASTNodeType::Update] =
-      std::bind(&ASTToOperatorTranslator::_translate_update_node, this, std::placeholders::_1);
-
-  // Maintenance operators
-  _operator_factory[ASTNodeType::ShowTables] =
-      std::bind(&ASTToOperatorTranslator::_translate_show_tables_node, this, std::placeholders::_1);
-  _operator_factory[ASTNodeType::ShowColumns] =
-      std::bind(&ASTToOperatorTranslator::_translate_show_columns_node, this, std::placeholders::_1);
-}
-
 std::shared_ptr<AbstractOperator> ASTToOperatorTranslator::translate_node(
```

Comment:
Should `ASTToOperatorTranslator::translate_node` make sure that no operator trees leave the translator that are not validated?